### ISSUE TITLE
Can `tailsamplingprocessor` be configured to take the "sampled" flag (from `trace-flags`) into account?

### ISSUE THREAD

**Issue by jpkrohling** [open-telemetry/opentelemetry-collector-contrib#32763]
### Discussed in https://github.com/open-telemetry/opentelemetry-collector-contrib/discussions/32495

<div type='discussions-op-text'>

<sup>Originally posted by **BatmanAoD** April 17, 2024</sup>
I have a service that is sending trace data to an OTEL collector, where it is sampled by the `tailsamplingprocessor` before being sent to Honeycomb. There is a client that does head-sampling, then sends its trace data directly to Honeycomb. I'd like to ensure that any calls that come from a client using head-sampling get sampled, i.e. that the `sampled` flag from the TraceContext's `trace-flags` is respected.

I can't tell if this is possible from the documentation of `tailsamplingprocessor`. @jpkrohling, in [this comment](https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/18462#issuecomment-1453419985), it sounds like you're saying that it's possible to use the `sampled` flag, but the README says:

> This processor must be placed in pipelines after any processors that rely on context, e.g. `k8sattributes`. It reassembles spans into new batches, causing them to lose their original context.

Does this mean that the `tailsamplingprocessor` doesn't receive the original trace context, including the `traceparent` header?</div>

**Comment by mikocot** [comment]
what is the blocker for this improvement? Is it conflicting with some existing changes or just needs additional cases to be taken into account?

**Comment by jpkrohling** [comment]
@mikocot, if you have a PR lined up, I'll review it.

**Comment by mikocot** [comment]
I don't, I'm not familiar with the code base, just wondering if that's something that is not implemented due to lack of resources, conflicting with some existing code, or it's not considered even the right approach.

**Comment by jpkrohling** [comment]
> lack of resources

Yes, lack of resources. I don't think there's anything that prevents us from taking the trace-flags into consideration.
